crates/ruff_python_formatter/src/statement/stmt_match.rs

```diff
@@ -1,12 +1,55 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
-use ruff_formatter::{write, Buffer, FormatResult};
+use ruff_formatter::{format_args, write, Buffer, FormatResult};
 use ruff_python_ast::StmtMatch;
 
+use crate::comments::dangling_comments;
+use crate::expression::maybe_parenthesize_expression;
+use crate::expression::parentheses::Parenthesize;
+use crate::not_yet_implemented_custom_text;
+use crate::prelude::*;
+use crate::{FormatNodeRule, PyFormatter};
+
 #[derive(Default)]
 pub struct FormatStmtMatch;
 
 impl FormatNodeRule<StmtMatch> for FormatStmtMatch {
     fn fmt_fields(&self, item: &StmtMatch, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let StmtMatch {
+            range: _,
+            subject,
+            cases,
+        } = item;
+
+        let comments = f.context().comments().clone();
+        let dangling_item_comments = comments.dangling_comments(item);
+
+        write!(
+            f,
+            [
+                text("match"),
+                space(),
+                maybe_parenthesize_expression(subject, item, Parenthesize::IfBreaks),
+                text(":"),
+                dangling_comments(dangling_item_comments)
+            ]
+        )?;
+
+        for case in cases {
+            write!(
```

Comment:
Do we need to retain empty lines between match cases?

Micha trying to write valid python code :sweat_smile: 

```python
match test:
	case "a":
		pass

	case "b":
		pass
```

Reply:
I think the newline handling will be done in the `MatchCase` implementation. Any newline between the `match` statement and the first case will be removed as is done in black as well:

```python
match test:

	case "a":
		pass

# vvvvv
match test:
	case "a":
		pass
```